x/observer/keeper/msg_server_disable_fast_confirmation.go

```diff
@@ -0,0 +1,56 @@
+package keeper
+
+import (
+	"context"
+
+	cosmoserror "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	authoritytypes "github.com/zeta-chain/node/x/authority/types"
+	"github.com/zeta-chain/node/x/observer/types"
+)
+
+// DisableFastConfirmation disables fast confirmation for the given chain ID
+// Inbound and outbound will be only confirmed using SAFE confirmation count on disabled chains
+func (k msgServer) DisableFastConfirmation(
+	goCtx context.Context,
+	msg *types.MsgDisableFastConfirmation,
+) (*types.MsgDisableFastConfirmationResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check permission
+	err := k.GetAuthorityKeeper().CheckAuthorization(ctx, msg)
+	if err != nil {
+		return nil, cosmoserror.Wrap(authoritytypes.ErrUnauthorized, err.Error())
+	}
+
+	// find current chain params list
+	chainParamsList, found := k.GetChainParamsList(ctx)
+	if !found {
+		return nil, types.ErrChainParamsNotFound
+	}
+
+	// disable fast confirmation
+	foundChain := false
+	for _, cp := range chainParamsList.ChainParams {
+		if cp.ConfirmationParams == nil {
+			return nil, types.ErrInvalidChainParams
+		}
```

Comment:
Let's add this check below when the chain is found

Reply:
updated in commit https://github.com/zeta-chain/node/pull/3522/commits/8f1370716f1d6714ef82a60dee38fa080adf9786